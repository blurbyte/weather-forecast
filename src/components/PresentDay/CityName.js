import styled from 'styled-components';

const CityName = styled.h1`
  color: ${props => props.theme.primary};
  font-family: 'Cardo', serif;
  line-height: 1;
  font-weight: 700;
  font-size: 4rem;
  letter-spacing: 2px;
  margin: 0;
`;

export default CityName;
