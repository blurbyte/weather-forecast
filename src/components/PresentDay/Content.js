import styled from 'styled-components';

import media from '../../styles/media';
import BaseContent from '../Content';

const Section = BaseContent.withComponent('section');

const Content = styled(Section)`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 4rem 2rem;

  ${media.phone`
    padding-top: 2rem;
    padding-bottom: 2rem;
  `};
`;

export default Content;
