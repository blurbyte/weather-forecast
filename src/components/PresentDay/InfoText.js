import styled from 'styled-components';

import BaseText from '../Text';

const InfoText = styled(BaseText)`
  margin: 0.2rem 2rem 0.2rem 0;
  text-transform: none;

  &:last-child {
    margin-right: 0;
  }
`;

export default InfoText;
