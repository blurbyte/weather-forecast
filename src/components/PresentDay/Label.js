import styled from 'styled-components';

const Description = styled.span`
  color: rgba(34, 34, 34, 0.5);
  margin-right: 0.4rem;
  text-transform: uppercase;
`;

export default Description;
