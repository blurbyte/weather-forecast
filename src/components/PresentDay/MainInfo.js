import React from 'react';
import styled from 'styled-components';

import BaseRow from '../Row';
import WeatherIcon from './WeatherIcon';
import Temperature from './Temperature';

const Row = styled(BaseRow)`
  margin-top: 0.4rem;
`;

const MainInfo = () => (
  <Row>
    <WeatherIcon />
    <Temperature>19&deg;</Temperature>
  </Row>
);

export default MainInfo;
