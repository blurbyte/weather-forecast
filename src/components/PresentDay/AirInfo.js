import React from 'react';
import PropTypes from 'prop-types';

import { removeDecimals } from '../../utils';
import Row from '../Row';
import InfoText from './InfoText';
import Label from './Label';

const AirInfo = ({ wind, humidity, pressure }) => (
  <Row>
    <InfoText>
      <Label>Wind</Label> {removeDecimals(wind)} m/s
    </InfoText>
    <InfoText>
      <Label>Humidity</Label> {humidity} %
    </InfoText>
    <InfoText>
      <Label>Pressure</Label> {removeDecimals(pressure)} hpa
    </InfoText>
  </Row>
);

AirInfo.propTypes = {
  wind: PropTypes.number.isRequired,
  humidity: PropTypes.number.isRequired,
  pressure: PropTypes.number.isRequired
};

export default AirInfo;
