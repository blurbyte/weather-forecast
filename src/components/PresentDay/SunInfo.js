import React from 'react';
import PropTypes from 'prop-types';

import { formatHour } from '../../utils';
import Row from '../Row';
import InfoText from './InfoText';
import Label from './Label';

const SunInfo = ({ sunrise, sunset }) => (
  <Row>
    <InfoText>
      <Label>Sunrise</Label> {formatHour(sunrise)}
    </InfoText>
    <InfoText>
      <Label>Sunset</Label> {formatHour(sunset)}
    </InfoText>
  </Row>
);

SunInfo.propTypes = {
  sunrise: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  sunset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired
};

export default SunInfo;
