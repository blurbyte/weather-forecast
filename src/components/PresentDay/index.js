import React from 'react';
import PropTypes from 'prop-types';

import isDaytime from '../../utils/isDaytime';
import Text from '../Text';
import Content from './Content';
import CityName from './CityName';
import MainInfo from './MainInfo';
import Weekday from './Weekday';
import SunInfo from './SunInfo';
import AirInfo from './AirInfo';

const PresentDay = ({ name, description, temp, sunrise, sunset, humidity, wind, pressure, weatherId }) => (
  <Content>
    <CityName>{name}</CityName>
    <Text big>{description}</Text>
    <MainInfo temp={temp} weatherId={weatherId} isDaytime={isDaytime(sunrise * 1000, sunset * 1000, Date.now())} />
    <Weekday timestamp={Date.now()} />
    <SunInfo sunrise={sunrise} sunset={sunset} />
    <AirInfo wind={wind} humidity={humidity} pressure={pressure} />
  </Content>
);

PresentDay.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  temp: PropTypes.number.isRequired,
  sunrise: PropTypes.number.isRequired,
  sunset: PropTypes.number.isRequired,
  humidity: PropTypes.number.isRequired,
  wind: PropTypes.number.isRequired,
  pressure: PropTypes.number.isRequired,
  weatherId: PropTypes.number.isRequired
};

export default PresentDay;
