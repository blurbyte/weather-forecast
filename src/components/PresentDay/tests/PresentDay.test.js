import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import MockDate from 'mockdate';
import moment from 'moment-timezone';

import { dayTheme } from '../../../styles/themes';
import PresentDay from '../index';

jest.mock('../../WeatherIcon', () => 'WeatherIcon');

test('renders correctly', () => {
  MockDate.set(moment('2018-05-19T18:00:00'));
  const props = {
    name: 'London',
    description: 'clear sky',
    temp: 12.34,
    sunrise: moment('2018-05-19T06:02:00').format(),
    sunset: moment('2018-05-19T20:54:00').format(),
    humidity: 45,
    wind: 2.4,
    pressure: 1024.24,
    weatherId: 800
  };

  const component = renderer.create(
    <ThemeProvider theme={dayTheme}>
      <PresentDay {...props} />
    </ThemeProvider>
  );
  expect(component.toJSON()).toMatchSnapshot();
  MockDate.reset();
});
