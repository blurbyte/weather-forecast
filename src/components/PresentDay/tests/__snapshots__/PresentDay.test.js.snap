// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-bxivhb cGzfDh sc-htpNat hClEhO"
>
  <h1
    className="sc-ifAKCX dZStwc"
  >
    London
  </h1>
  <p
    className="sc-bdVaJa hPriLU"
  >
    clear sky
  </p>
  <div
    className="sc-gzVnrw bsdMms sc-EHOje leSkOi"
  >
    <WeatherIcon
      isDaytime={false}
      weatherId={800}
    />
    <p
      className="sc-bZQynM laSBUb"
    >
      12
      Â°
    </p>
  </div>
  <div
    className="sc-dnqmqq eexdWW sc-EHOje leSkOi"
  >
    <p
      className="sc-bdVaJa hPriLU"
    >
      <span
        className="sc-htoDjs loeAKY"
      >
        Today
      </span>
       
      Saturday
    </p>
  </div>
  <div
    className="sc-EHOje leSkOi"
  >
    <p
      className="sc-iwsKbI drgGaZ sc-bdVaJa jxXTGO"
    >
      <span
        className="sc-htoDjs loeAKY"
      >
        Sunrise
      </span>
       
      06:02
    </p>
    <p
      className="sc-iwsKbI drgGaZ sc-bdVaJa jxXTGO"
    >
      <span
        className="sc-htoDjs loeAKY"
      >
        Sunset
      </span>
       
      20:54
    </p>
  </div>
  <div
    className="sc-EHOje leSkOi"
  >
    <p
      className="sc-iwsKbI drgGaZ sc-bdVaJa jxXTGO"
    >
      <span
        className="sc-htoDjs loeAKY"
      >
        Wind
      </span>
       
      2
       m/s
    </p>
    <p
      className="sc-iwsKbI drgGaZ sc-bdVaJa jxXTGO"
    >
      <span
        className="sc-htoDjs loeAKY"
      >
        Humidity
      </span>
       
      45
       %
    </p>
    <p
      className="sc-iwsKbI drgGaZ sc-bdVaJa jxXTGO"
    >
      <span
        className="sc-htoDjs loeAKY"
      >
        Pressure
      </span>
       
      1024
       hpa
    </p>
  </div>
</section>
`;
